src/clinical/clinical-service.ts

```diff
@@ -448,56 +467,59 @@ export const getValidRecordsPostSubmission = async (
 					.filter(filterDuplicates);
 
 				const donorErrorRecords = await Promise.all(
+					// ... group errors by Entity, then filter stale error records
 					currentDonorEntities.map(async (entityName) => {
 						const clinicalRecords: ClinicalInfo[] = getClinicalEntitiesFromDonorBySchemaName(
 							currentDonor,
 							entityName,
 						);
 
-						const stringifiedRecords = clinicalRecords
+						let filteredErrors = [...currentDonorErrors];
```

Comment:
can probably use more than one array instead of reassigning `filteredErrors` 3 times.
favour the developer experience benefit unless we know this is an optimisation we need to make due to a recorded bottleneck